addons/io_scene_gltf2/blender/exp/gltf2_blender_extract.py

```diff
@@ -537,18 +537,10 @@ def __get_uvs(blender_mesh, uv_i):
 
 
 def __get_colors(blender_mesh, color_i):
-    layer = blender_mesh.vertex_colors[color_i]
     colors = np.empty(len(blender_mesh.loops) * 4, dtype=np.float32)
-    layer.data.foreach_get('color', colors)
+    blender_mesh.color_attributes[color_i].data.foreach_get('color', colors)
```

Comment:
Actually color_i is an index into the `mesh.vertex_colors` list. You should keep the layer variable and use `blender_mesh.color_attributes[layer.name]` to get the data. `color_attributes` may have more stuff that isn't supported yet.